woocommerce/woocommerce-paypal-payments
# Add Smart Button Language setting for PayPal buttons to override automatic language detection

**Issue**: #600

---

### Description

Some merchants are asking for a way to only display the PayPal interface and buttons in one fixed locale.

The locale renders components. By default PayPal smartly detects the correct locale for the buyer based on their geolocation and browser preferences. It is recommended to pass this parameter only if you need the PayPal buttons to render in the same language as the rest of your site.

The PR will:

- Add a new setting to select the language for PayPal smart buttons. 
- Add a filter to override the available locales: `woocommerce_paypal_payments_button_locales`
- Add a filter to change/override the selected locale: `woocommerce_paypal_payments_smart_buttons_locale`

### Steps to Test

1. Activate PayPal button on any location.
2. Enable tagline and/or button label.
3. Set the "Smart Button Language" setting.
4. Check the live previews.
7. Visit page with buttons.
8. → button label should be the selected language.

---

Closes #600 .
